module/x/gravity/keeper/batch.go

```diff
@@ -185,6 +186,34 @@ func (k Keeper) GetUnslashedOutgoingTxs(ctx sdk.Context, maxHeight uint64) (out
 	return
 }
 
+func (k Keeper) GetUnconfirmedBatchTxs(ctx sdk.Context, val sdk.ValAddress) []*types.BatchTx {
+	var unconfirmed []*types.BatchTx
+	k.IterateCompletedOutgoingTxsByType(ctx, types.BatchTxPrefixByte, func(_ []byte, cotx types.OutgoingTx) bool {
+		sig := k.getEthereumSignature(ctx, cotx.GetStoreIndex(), val)
+		if len(sig) == 0 { // it's pending
```

Comment:
Nit: I've noticed this comment elsewhere, but completed outgoing TXs aren't "pending".